app/src/main/java/com/example/myapplication/activity/MainActivity.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.myapplication.activity
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import com.example.myapplication.R
+import com.example.myapplication.mvp.model.CalculatorModel
+import com.example.myapplication.mvp.presenter.CalculatorPresenter
+import com.example.myapplication.mvp.view.CalculatorView
+import kotlinx.android.synthetic.main.activity_main.button_divide
+import kotlinx.android.synthetic.main.activity_main.button_eight
+import kotlinx.android.synthetic.main.activity_main.button_equal
+import kotlinx.android.synthetic.main.activity_main.button_five
+import kotlinx.android.synthetic.main.activity_main.button_four
+import kotlinx.android.synthetic.main.activity_main.button_less
+import kotlinx.android.synthetic.main.activity_main.button_multiplication
+import kotlinx.android.synthetic.main.activity_main.button_nine
+import kotlinx.android.synthetic.main.activity_main.button_one
+import kotlinx.android.synthetic.main.activity_main.button_plus
+import kotlinx.android.synthetic.main.activity_main.button_point
+import kotlinx.android.synthetic.main.activity_main.button_seven
+import kotlinx.android.synthetic.main.activity_main.button_six
+import kotlinx.android.synthetic.main.activity_main.button_three
+import kotlinx.android.synthetic.main.activity_main.button_two
+import kotlinx.android.synthetic.main.activity_main.button_zero
+
+class MainActivity : AppCompatActivity(), View.OnClickListener {
+
+    private var presenter : CalculatorPresenter ?= null
```

Comment:
dont make the presenter nullable. 
initialize it right away

Reply:
done